File: tests/src/test/java/com/optivem/atddaccelerator/templategenerator/systemtest/dsl/github/helpers/RepositoryClient.java
package com.optivem.atddaccelerator.templategenerator.systemtest.dsl.github.helpers;

import com.optivem.atddaccelerator.templategenerator.systemtest.clients.GithubClient;

import static com.optivem.atddaccelerator.templategenerator.systemtest.util.ProcessResultAssertions.assertSuccess;

public class RepositoryClient {
    private final GithubClient client;
    private final String repositoryPath;

    public RepositoryClient(GithubClient client) {
        this.client = client;
        this.repositoryPath = client.getRepositoryPath();
    }

    public void verifyRepositoryExists() {
        var result = client.viewRepository();
        assertSuccess(result, "Repository '" + client.getRepositoryPath() + "' should exist.");
    }

    public void deleteRepository() {
        var result = client.deleteRepository();
        assertSuccess(result, "Failed to delete repository '" + client.getRepositoryPath() + "'.");
    }
}

File: console/tests/Dsl/GitHub/Helpers/RepositoryClient.cs
using Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Clients;
using static Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Util.Process.ProcessResultAssertions;

namespace Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Dsl.GitHub.Helpers
{
    public class RepositoryClient
    {
        private readonly GithubClient _client;
        private readonly string _repositoryPath;

        public RepositoryClient(GithubClient client)
        {
            _client = client;
            _repositoryPath = client.GetRepositoryPath();
        }

        public void VerifyRepositoryExists()
        {
            var result = _client.ViewRepository();
            AssertSuccess(result, $"Repository '{_client.GetRepositoryPath()}' should exist.");
        }

        public void DeleteRepository()
        {
            var result = _client.DeleteRepository();
            AssertSuccess(result, $"Failed to delete repository '{_client.GetRepositoryPath()}'.");
        }
    }
}